### ISSUE TITLE
[FEAT] Create a util custom layer with Gumbel-Softmax for one categorical distribution

### ISSUE THREAD

**Issue by jfsantos-ds** [ydataai/ydata-synthetic#122]
A Gumbel-Softmax layer works on logits or probability outputs of a model over the distribution of a categorical feature.
The required computations of this layer's forward method are the following:
- Soft sample: Softmax of a tensor consisting of a Gumbel sample added to logits. The gradients of this computation are stored.
- Hard sample: A one-hot encoded categorical variable with a sampled class activated. The gradients of this computation are ***not*** stored, but used only for obtaining _real_ categorical samples.

Note: Generally only the hard sample should be returned in the forward call, alternatively both can be returned in an (hard, soft) tuple.

Requirements:
- TF implementation
- A standalone custom layer implemented in utils that produces the requested output assuming as input logits from a single categorical distribution
